### PR TITLE
Bump terraform run version constraint

### PR DESCRIPTION
**Pull request by hajapy**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Closes https://github.com/conda-forge/terragrunt-feedstock/issues/87

I chose to set the specific constraint to match the compatibility matrix: https://terragrunt.gruntwork.io/docs/getting-started/supported-terraform-versions/, though we could also consider a more flexible range.

### CODE REVIEW

**Comment by hajapy** [comment]
@conda-forge-admin please rerender

**Review by asad-awadia — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,15 +12,15 @@ source:
...
+  number: 1
   script:
     - go install -v -ldflags "-X main.VERSION={{ version }}" .
 
 requirements:
   build:
     - {{ compiler('go') }}
   run:
-    - terraform 0.12
+    - terraform 0.13
```
Comment (hajapy) [inline recipe/meta.yaml:23]: I just noticed terragrunt 0.27 should also work with terraform 0.14 per the release notes: https://github.com/gruntwork-io/terragrunt/releases/tag/v0.27.0

Should this be a range or unpinned and let people rely on eg. https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#terraform_version_constraint to set their desired terraform version?

Comment (hajapy) [reply recipe/meta.yaml]: I've opted for a range of terraform versions with known support rather than unpinned, as it seems like the safer choice.

**Comment by hajapy** [comment]
@mariusvniekerk @sodre ready for review

**Comment by hajapy** [comment]
@conda-forge/core could someone have a look at this?

**Comment by beckermr** [comment]
We wait a week for these requests. So bump us again then. Hopefully someone from @conda-forge/terragrunt can get to it sooner.

**Comment by hajapy** [comment]
@conda-forge-admin please rerender

**Comment by mariusvniekerk** [comment]
Thanks @hajapy !

**Pull request merged**